### DISCUSSION TITLE
Getting log info for debug

### DISCUSSION THREAD

**Discussion by robbrad** [frenck/action-home-assistant#57] [General]
@frenck - great work here buddy.

I have a custom componenet that works on my local install but not on the pipleine - is there anyway to get the debug logs from the action run of HA check config?

Failed config
  General Errors: 
    - Platform error image_processing.classificationbox - Integration 'classificationbox' not found.

**Comment by frenck** [comment]
The only way to resolve that / making the Home Assistant configuration check work; is by adding/committing the custom integration into the repository.

Alternatively, you could adjust your GitHub Actions workflow to download and install the custom integration into place from its source repository before starting the Home Assistant configurations check.

Without the integration present, the configuration check will fail, because well, the referenced integration doesn't exist.

**Reply by robbrad** [reply]
That’s exactly what I do - I download the custom component from the requisite repo before the check

https://github.com/robbrad/HA-Config/blob/master/.github/workflows/main.yml

There is some other issue prob with the custom component that stops it working on the pipeline but does locally. Hence the debug idea on the pipeline

**Reply by frenck** [reply]
Maybe I'm missing the point?

Or are you suggesting to try to catch the logs and annotate them in GitHub Actions?

**Reply by robbrad** [reply]
Yes that is what I’m suggesting- or make it an optional verbosity

**Reply by frenck** [reply]
Something like this?

https://github.com/frenck/action-home-assistant/pull/58

**Reply by robbrad** [reply]
Exactly like that - will that pull out specifics of why the intergration wasnt found - Want me to test it for you?

**Reply by frenck** [reply]
There is no specific on "why" it wasn't found. As in, it simply doesn't exist.

I will pull out the issue from the logs, and highlight it in GitHub actions using an annotation (so it becomes more visible).

**Reply by robbrad** [reply]
Ok - so its already quite visable see:  

![image](https://user-images.githubusercontent.com/6122503/158150589-c93cb478-1311-4d63-aa0e-74aa526a1b10.png)

The issue I have (which is strange) is it works for my local install but not for the pipeline - but obviously not got an easy way here to crack open why its not working on the pipeline

The custom componenent code is pulled in the same way locally as it is on the pipeline - (code base is the same) so its got to be some other viarable im not considering

I might try to do the "check" locally the same way the pipeline does to see if that uncovers anything strange

**Reply by robbrad** [reply]
Something very odd going on with that custom component - I lack the experiance to debug it unfortunatly  https://github.com/robmarkcole/HASS-Machinebox-Classificationbox/tree/master/custom_components/classificationbox

When running python3 -m homeassistant --config ../../.homeassistant/ --script check_config from my HA venv I get the same error as the pipeline 

Failed config
  General Errors:
    - Platform error image_processing.classification_box - Integration 'classification_box' not found.

Successful config (partial)

However the componenet is found and works when HA uses it for real

I do have another imageprocessing custom componenet and worried they might clash

So my options are to keep hacking around with the custom componenet code or work out how I cant debug why **--script check_config** is failing to find the intergration

**Reply by robbrad** [reply]
Found it !  - sorry to waste your time @frenck 

2022-03-14 11:38:25 ERROR (SyncWorker_5) [homeassistant.loader] The custom integration 'classificationbox' does not have a version key in the manifest file and was blocked from loading. See https://developers.home-assistant.io/blog/2021/01/29/custom-integration-changes#versions for more details

**Reply by frenck** [reply]
Don't worry! Stuff happens :) At least I got a nice little annotation out of it :)

**Reply by robbrad** [reply]
I feel like though this discovery the —script check_config should have a more verbose output. 

I only found that issue buried in my runtime logs.

**Reply by frenck** [reply]
Hmm yeah, I guess the main issue is: It is not really config that fails at this point, but an integration 😕

**Reply by robbrad** [reply]
Feels like if https://github.com/home-assistant/core/blob/dev/homeassistant/scripts/check_config.py#L231 put out errors that would be helpful to the check config call
